### ISSUE TITLE
Fatal error: Null provided to count() in InheritanceAnalyzer

### ISSUE THREAD

**Issue by vasanth-kumar-m-y** [pdepend/pdepend#875]
- PHP Depend version: 2.5.0
- PHP Version: 8.1.28
- Installation type: (phar file)
- Operating System / Distribution & Version: (CentOS 7.6)

### Current Behavior
Getting this error when running **phpmd-extension** which has dependency on **pdepend/pdepend 2.5.0**

`PHP Fatal error:  Uncaught TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/pdepend/pdepend/src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php:204`

### Expected Behavior
Should run phpmd without any issues

**Comment by vasanth-kumar-m-y** [comment]
@tvbeek  @kylekatarnls  could you please guide here

**Comment by AJenbo** [comment]
@vasanth-kumar-m-y could you please test if this issue is fixed in latest test build of PHPMD 3.0, you can get the Phar directly here: https://github.com/phpmd/phpmd/actions/runs/9447536647/artifacts/1585146970

Also you seam to have skipped on filling out most of the template for reporting an issue.

**Comment by vasanth-kumar-m-y** [comment]
Thanks for the quick response @AJenbo 

Yes i tried with the latest version **phpmd/phpmd 3.x-dev** still the same issue. Getting this issue after upgrading symfony 6.4 and PHP version 8.1.28.

I am not sure how to use this phar file as the pdepend/pdepend is installed as dependency of phpmd/phpmd.

**Comment by AJenbo** [comment]
I updated the link to be the phar for phpmd instead, at first i had linked pdepends since I thought that was what you where having issues with.

Can you please provide a link to the code base that is triggering this issue, or a code sample?

If you still get the issue with the latest 3.0-dev then can you please provide the line numbers it reports as things have moved around between the two versions.

**Comment by vasanth-kumar-m-y** [comment]
@AJenbo  
In composer.json

```
"require-dev": {
       "phpmd/phpmd": "^2.15",
       "tm/tooly-composer-script": "^1.4",
},
"tools": {
            "phpmd-extension": {
                "url": "https://github.com/mi-schi/phpmd-extension/releases/download/stable/phpmd-extension.phar",
                "only-dev": true,
                "force-replace": false
            }
        }
```
        
PHPMD command 
`/vagrant/bin/phpmd-extension.phar src text .phpmd.xml`

resulting in 

```
PHP Fatal error:  Uncaught TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/pdepend/pdepend/src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php:204
Stack trace:
#0 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/pdepend/pdepend/src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php(204): count()
#1 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/pdepend/pdepend/src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php(182): PDepend\Metrics\Analyzer\InheritanceAnalyzer->doAnalyze()
#2 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php(608): PDepend\Metrics\Analyzer\InheritanceAnalyzer->analyze()
#3 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/pdepend/pdepend/src/main/php/PDepend/Engine.php(331): PDepend\Engine->performAnalyzeProcess()
#4 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/phpmd/phpmd/src/main/php/PHPMD/Parser.php(123): PDepend\Engine->analyze()
#5 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/phpmd/phpmd/src/main/php/PHPMD/PHPMD.php(222): PHPMD\Parser->parse()
#6 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php(129): PHPMD\PHPMD->processFiles()
#7 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/phpmd/phpmd/src/main/php/PHPMD/TextUI/Command.php(173): PHPMD\TextUI\Command->run()
#8 phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/bin/phpmd(121): PHPMD\TextUI\Command::main()
#9 /vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar(10): require('...')
#10 {main}
  thrown in phar:///vagrant/vendor/tm/tooly-composer-script/bin/phpmd-extension.phar/vendor/pdepend/pdepend/src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php on line 204
```

**Comment by AJenbo** [comment]
Ok, the explanation here is that you are using a tool based on [2.5.0](https://github.com/pdepend/pdepend/releases/tag/2.5.0) an 8 year old release.

The issue you are reporting was fixed 7 years ago in https://github.com/pdepend/pdepend/pull/303 and included in the following patch release of 2.5.1.

Please report the issue with [phpmd-extension](https://github.com/mi-schi/phpmd-extension) instead. All they need to do is update there dependencies. But note that they have not updated there project in 6 years. Is it possible that the relevant rules are now part of the regular PHPMD releases? Maybe try switching to it instead.

**Comment by vasanth-kumar-m-y** [comment]
Thanks for your time and looking into this, will check.
